src/builtin.c

```diff
@@ -363,6 +363,33 @@ static jv f_tonumber(jq_state *jq, jv input) {
   return type_error(input, "cannot be parsed as a number");
 }
 
+static jv f_toint(jq_state *jq, jv input) {
+  input = f_tonumber(jq, input);
+  if (!jv_is_valid(input))
+    return input;
+  if (jv_is_int64(input) || jv_is_uint64(input))
+    return input;
+  double d = jv_number_value(input);
+  if (d < 0 && d >= INT64_MIN) {
+    int64_t i = d;
+    if (i < 0)
```

Comment:
Perhaps I misunderstand.
Line 373 is checking to see if `d is negative AND d fits into an int64_t`
Line 374 does the cast
What does line 375 check?